framework/src/media/InputHandler.cpp

```diff
@@ -70,23 +71,6 @@ bool InputHandler::doStandBy()
 	return true;
 }
 
-bool InputHandler::open()
-{
-	/* Media f/w playback supports only mono and stereo.
-	 * In case of multiple channel audio, we ask decoder always outputting stereo PCM data.
-	 */
-	unsigned int channels = getDataSource()->getChannels();
```

Comment:
StreamHandler::open() -> InputDataSource::open() -> then we can get correct channel info.
Before `open()`, here getChannels() always return the default 2.
So I move this code to InputHandler::registerCodec(...)